### DISCUSSION TITLE
resiliency4j with reactor: transformDeferred or transform

### DISCUSSION THREAD

**Discussion by trung** [resilience4j/resilience4j#1829] [Q&A]
Hi, I saw that the example uses `transformDeferred(RetryOperator.of(..))` instead of `transform(...)`.
My thinking is that: with `transformDeferred()`, retry operator will be applied separately to individual subscriber. Whereas `transform()` will share the retry operator among subscribers. 

For example, my reactive API returns a `Flux`. If this `Flux` is being resubscribed across requests then `transformDeferred()` makes sense, otherwise if `Flux` is created for each request, then `transform()` would be ok.

My question is:

1. Is my understanding above correct?
2. In practice, will the reactive-enabled server resubscribe the `Flux` across multiple requests? Or will it create a new `Flux` for each request?

Many thanks,

Trung

**Comment by RobWin** [comment]
Hi,
I think this question is better raised in the Reactor GitHub repo.

**Reply by RassulYunussov** [reply]
Hi @RobWin!
I think the question is valid and it would be good to have a thorough explanation why that method is preferred instead of a transform.
As within documentation it is explicitly demonstrated, that transformDeferred is used everywhere: https://resilience4j.readme.io/docs/examples-1

I'd love to see that explanation. 
Thank you!
